### PR TITLE
Added configurable max limit for branch retrieval client

### PR DESCRIPTION
**Pull request by jaguilar-atl**
* Added system properties for allowing a configurable max limit for the branch retrieval client.
  - `bitbucket.remote.branches.retrieval.max.pages` - (defaults to 5)
  - `bitbucket.remote.branches.retrieval.page.size` - (defaults to 1000)
* Modified `BitbucketBranchClientImpl` to use the configurable limits.

### CODE REVIEW

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketBranchClientImpl.java [hunk 1]
```diff
@@ -36,34 +41,53 @@ public Stream<BitbucketDefaultBranch> getRemoteBranches() {
...
         }
 
         @Override
         public BitbucketPage<BitbucketDefaultBranch> next(BitbucketPage<BitbucketDefaultBranch> previous) {
             if (previous.isLastPage()) {
                 throw new IllegalArgumentException("Last page does not have next page");
             }
+
+            if (currentPage.incrementAndGet() >= maxPages) {
+                // We've reached the maximum number of pages, so we return an "empty last page" to stop the iterator.
```
Comment (jaguilar-atl) [inline src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketBranchClientImpl.java:80]: Add logging to inform users that the limit has been met.

Comment (dkjellin) [reply src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketBranchClientImpl.java:80]: Depending on how you feel about it, you could modify the `BitbucketPage` to contain the information needed for logging. (I don't know if this is a good idea, maybe it is just an ugly hack)

**Comment by dkjellin** [comment]
I think this looks good. Like you already mention I think it would be good to add some logging.

File: src/test/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketBranchClientImplTest.java [hunk 2]
```diff
@@ -26,7 +26,7 @@
 @RunWith(MockitoJUnitRunner.class)
 public class BitbucketBranchClientImplTest {
 
-    private static final String BRANCHES_URL = "%s/rest/api/1.0/projects/%s/repos/%s/branches?limit=1000";
+    private static final String BRANCHES_URL = "%s/rest/api/1.0/projects/%s/repos/%s/branches?limit=1000&orderBy=modification";
```
[the remark this answers is not in the document]
Comment (dkjellin) [reply src/test/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketBranchClientImplTest.java:29]: It is case insensitive, the REST docs should perhaps be updated to mention it is case insensitive.

**Review by dkjellin — APPROVED** [review]

**Pull request merged**